src/test/java/teammates/test/pageobjects/InstructorCourseStudentDetailsEditPage.java

```diff
@@ -73,10 +73,19 @@ private void fillStudentDetailsForm(String studentName, String teamName, String
             fillTextBox(commentsTextbox, comments);
         }
         if (teamName == null) {
-            click(submitButton);
+            if (studentEmail == null) {
+                click(submitButton);
+            } else {
+                clickAndConfirm(submitButton);
+            }
+            
         } else {
             // if team name is edited, the confirmation dialog will pop up
-            clickAndConfirm(submitButton);
+            if (studentEmail == null) {
+                clickAndConfirm(submitButton);
+            } else {
+                clickAndConfirmTwoTimes(submitButton);
```

Comment:
Take a look at my previous comment. If anything, it is because this method looks hackish and too retrofitted for this specific purpose.